src/embedded.ts

```diff
@@ -284,7 +284,11 @@ export class RoomWidgetClient extends MatrixClient {
                 const rawEvents = await this.widgetApi.readStateEvents(eventType, undefined, stateKey, [this.roomId]);
                 const events = rawEvents.map((rawEvent) => new MatrixEvent(rawEvent as Partial<IEvent>));
 
-                await this.syncApi!.injectRoomEvents(this.room!, [], events);
+                if (this.syncApi instanceof SyncApi) {
+                    await this.syncApi.injectRoomEvents(this.room!, undefined, events);
+                } else {
+                    await this.syncApi!.injectRoomEvents(this.room!, events);
+                }
```

Comment:
This would be nice to get a comment explaining:
What is v2 what is sliding sync?
What is the second what is the third arg?
What does `injectRoomEvents` room do in the different overloaded versions?

(This is also explained below `onEvent` but i think duplicating it here makes sense)